### PR TITLE
Patch check shapeshit

### PR DESCRIPTION
**Pull request by NikoCat233**
Patch checkshapeshift
so some roles with 1 ss duration wont have to suffer the pain of useless shape shift

### CODE REVIEW

**Comment by NikoCat233** [comment]
Gonna support the roles i changed in part 1 first.
Some roles have their own ss duration so I will leave it later.

File: Roles/Impostor/RiftMaker.cs [hunk 1]
```diff
@@ -113,7 +113,8 @@ public static void ReceiveRPC(MessageReader reader)
     public static void ApplyGameOptions()
     {
         AURoleOptions.ShapeshifterCooldown = SSCooldown.GetFloat();
-        AURoleOptions.ShapeshifterLeaveSkin = true;
+        //AURoleOptions.ShapeshifterLeaveSkin = true;
+        //You set leave skin here only influences the rifter it own. Its not global
         AURoleOptions.ShapeshifterDuration = 1f;
```
Comment (Tommy-XL) [inline Roles/Impostor/RiftMaker.cs:118]: What? It should be because for Rift Maker its helps a lot to determine exactly where he placed the marker.

Comment (NikoCat233) [reply Roles/Impostor/RiftMaker.cs:118]: huh, but if we reject the shapeshift animation, there wont be skin left

Comment (Tommy-XL) [reply Roles/Impostor/RiftMaker.cs:118]: Bad

**Comment by NikoCat233** [comment]
Currently these roles' shapeshift animation is hidden:
Undertaker, EvilTracker, Hacker, Bomber, Nuker, QuickShooter, Dazzler, Devourer, Blackmailer, RiftMaker, Pitfall

these roles' shapeshift calls is rejected bcz their ss button is used to display skill timer:
Penguin, BountyHunter, SerialKiller

Roles that have their own shapeshift duration settings will not be reworked in this pr.

We need to make sure the roles mentioned above works fine with the reject shapeshift technique first, then we can rework more roles. Pls roll out a build to test the roles above.

**Comment by Tommy-XL** [comment]
I think it's better to leave this for 1.6.0 (after release)

**Comment by NikoCat233** [comment]
OK

**Comment by Ultradragon005** [comment]
This works for vanilla? I remember something else #224 like this being made and #224 didn't work for vanilla

**Comment by Tommy-XL** [comment]
Should be working
image

**Comment by Tommy-XL** [comment]
`player.RpcRejectShapeshift()` As far as I understand, it completely cancels the effect of shapeshift

**Comment by Ultradragon005** [comment]
yeah I just noticed that previous pr I mentione didn't use that.. so great then

**Comment by Tommy-XL** [comment]
Yeah, vanilla players froze in place because `player.waitingForShapeshiftResponse` return `true`
image

**Comment by Gurge44** [comment]
I've done this months ago in TOHE+ and it works perfectly.
RejectShapeShift just sets waitingForShapeshiftResponse to false

**Pull request merged**